utility/NonceManage.ts

```diff
@@ -0,0 +1,53 @@
+import { ethers } from 'ethers';
+import { NonceManager } from '@ethersproject/experimental';
+import { logger } from '../utility/Logger';
+
+
+const _logger = logger.getLogger( 'NonceManage');
+
+export class NonceManage extends NonceManager {
+  private _pending: number = 0;
+  private _waited: number = 0;
+  private _threshold: number = 2;
+
+  constructor( signer: ethers.Signer, 
+                threshold?: number) {
+    super( signer);
+    if (threshold)
+      this._threshold = threshold;
+  }
+
+
+  async sendTransaction( txn: ethers.utils.Deferrable< 
+                                        ethers.providers.TransactionRequest>) : 
+                          Promise< ethers.providers.TransactionResponse>  {
+    _logger.debug( `super send with ${this._pending} pending`);
+
+    if (this._pending >= this._threshold) {
```

Comment:
> Isn't this whole block the same as doing
> 
> `while (this._pending >= this._threshold) { await sleep(1000); }`
> 
> That is more easily understood

No it's not the same, and actually this is the simplest construction I found after considering a number of approaches. Have you tried any of this yourself or are you mentally projecting?

If you had not banned PDL, I would have clarified the logic (and so shown its efficiency) that way. If you like, I can add it in for this specific method since it seems to be troublesome.

Basically if a txn gets to this method but too many are already in the pending queue actively being processed, the txn needs to be placed into the waiting pool instead. Every second, the txn checks if a slot has opened up, and if it has, it switches itself into the pending queue and must *immediately* increment the pending counter. The logging statements are also useful/necessary to debug/clarify operation if the need ever arises.

Really, I see this as simple and clean, and don't understand how or why you'd consider it complicated. It's doing only what it needs to do with nothing extraneous or inefficient?